TRB_label_assist/models.py

```diff
@@ -880,13 +887,19 @@ def set_params(self, params):
 
         return self
 
-    def fit(self, train_df):
+    def fit(self, train_df,ct_entry):
         # fit clustering model
-        self.end_cluster_model.fit(train_df)
+        if self.__class__.__name__ == 'RefactoredNaiveCluster':
```

Comment:
passing the `Entry' type data to `fit` function of `RefactoredNaiveCluster`. Since there are other models as well ( with their own fit), changing the `fit` without if-else would mean we would have change fit functions definitions of  all involved models and pass `ct_entry` to these even when they don't  need it.

Reply:
I see what you say. However, in terms of clean interfaces, I think it would actually be better to have all the `fit` methods take in the same set of parameters, so that we can run them interchangeably and compare the results. 

Think of `sklearn` - does it have separate interfaces for each type of ML model? No.
One of its many great strengths is that it provides a standard interface, so you can write something like 

```
for model in list_of_models:
    model.fit(X, y)
    ...
```

That is what we are going for here. I know that you said that you haven't run `performance_eval.py` yet, but you might want to at least look at what it expects. 

It essentially does something very similar

```
        model_ = model()
        if model_params is not None:
            model_.set_params(model_params)
        train_trips = user_df.iloc[train_idx]
        test_trips = user_df.iloc[test_idx]

        # train the model
        logging.info("About to fit the model %s" % model)
        model_.fit(train_trips)
```

I thought that Hannah did a great job of putting a nice, generic interface on top of all the models that we compare in the paper

Reply:
yeah that makes sense.